modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/layer/SymbolMBLayer.java

```diff
@@ -1732,10 +1746,15 @@ public Expression textTranslateAnchor() {
                 "symbol-spacing", getId());
         symbolizer.getOptions().put("repeat", String.valueOf(symbolSpacing));
 
-        // text max angle
-        // layer.getTextMaxAngle();
-        // symbolizer.getOptions().put("maxAngleDelta", "40");
-
+        // text max angle - only for line placement
+        // throw MBFormatException if point placement
+        if(labelPlacement instanceof LinePlacement){
+            // will be true if line placement, false else.
+            symbolizer.getOptions().put("followLine", String.valueOf(followLine));
+            symbolizer.getOptions().put("maxAngleDelta", String.valueOf(getTextMaxAngle()));
+        }else if(labelPlacement instanceof PointPlacement && hasTextMaxAngle()){
```

Comment:
`labelPlacement instanceof PointPlacement` is redundant. Just need `} else if (hasTextMaxAngle()) {`

Also, take a look at your formatting - there should be spaces as in my example above.